src/python/WMCore/WMSpec/WMTask.py

```diff
@@ -1355,13 +1362,36 @@ def setNumberOfCores(self, cores, nStreams):
 
         return
 
-    def setAcquisitionEra(self, era, parentAcquisitionEra=None):
+    def _getStepValue(self, keyDict):
+        """
+        __getStepValue_
+
+        Maps this taskName - in somehow a hacky way - to a 'StepName' value
+        that should exist in a StepChain request. Used only on tasks that have
+        output module
+        :param keyDict: a dict with either AcqEra/ProcStr/ProcVer key/value pairs,
+        where the key corresponds to the StepName
+        """
+        value = None
+        if self.taskType() == "Merge":
+            extractedTaskName = self.name().split("Merge")[0]
+            value = keyDict.get(extractedTaskName)
+        elif self.taskType() in ["Production", "Processing"]:
+            value = keyDict.get(self.name())
+
+        return value
+
```

Comment:
That's correct! Have a look at the unit test example where it lists all task names and you can clearly see how the StepName value is mapped to an actual task (but in the end, what you said is correct ;)